rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1017,7 +1017,41 @@ private void logUpdateCollections(Transaction rskTx) {
         eventLogger.logUpdateCollections(sender);
     }
 
-    protected void processSVPFailure(Federation proposedFederation) {
+    protected void updateSvpState(Transaction rskTx) {
+        Optional<Federation> proposedFederationOpt = federationSupport.getProposedFederation();
+        if (proposedFederationOpt.isEmpty()) {
+            return;
+        }
+
+        Keccak256 rskTxHash = rskTx.getHash();
+        // if the proposed federation exists and the validation period ended,
+        // we can conclude that the svp failed
+        Federation proposedFederation = proposedFederationOpt.get();
+        if (!validationPeriodIsOngoing(proposedFederation)) {
+            processSVPFailure(proposedFederation);
+            return;
+        }
+
+        // if the fund tx signed is present, then the fund transaction change was registered,
+        // meaning we can create the spend tx.
+        Optional<BtcTransaction> svpFundTxSigned = provider.getSvpFundTxSigned();
+        if (svpFundTxSigned.isPresent()) {
+            processSvpSpendTransactionUnsigned(rskTxHash, proposedFederation, svpFundTxSigned.get());
+            return;
+        }
+
+        // if the unsigned fund tx hash is not present, meaning we can proceed with the fund tx creation
+        Optional<Sha256Hash> svpFundTxHashUnsigned = provider.getSvpFundTxHashUnsigned();
+        if (svpFundTxHashUnsigned.isEmpty()) {
+            try {
+                processSvpFundTransactionUnsigned(rskTxHash, proposedFederation);
+            } catch (Exception e) {
+                logger.error("[updateSvpState] Error processing svp fund transaction unsigned.");
+            }
+        }
```

Comment:
I was thinking on how we can change the order of these two logic blocks get executed. I think it would be easier to understand if the logic related to processing `svpFundTxHashUnsigned` happens first than the `svpFundTxSigned`. Maybe we can do something like this:

```java
// Check if the unsigned fund transaction hash is not present, and create the fund transaction if needed
Optional<Sha256Hash> svpFundTxHashUnsigned = provider.getSvpFundTxHashUnsigned();
if (svpFundTxHashUnsigned.isEmpty()) {
    try {
        processSvpFundTransactionUnsigned(rskTxHash, proposedFederation);
    } catch (Exception e) {
       ...
    }
   return;
}

// Check if the signed fund transaction is present, and create the spend transaction if available
Optional<BtcTransaction> svpFundTxSigned = provider.getSvpFundTxSigned();
if (svpFundTxSigned.isPresent()) {
    processSvpSpendTransactionUnsigned(rskTxHash, proposedFederation, svpFundTxSigned.get());
}
```

Reply:
remember that when we save the `svpFundTxSigned` we remove the `svpFundTxHashUnsigned`. So this order won't have the expected behavior

Reply:
updated